pkg/apis/internal.admin.acorn.io/v1/resources.go

```diff
@@ -75,86 +128,78 @@ func (current *Resources) Remove(incoming Resources, all bool) {
+		return fmt.Errorf("%w: %s", ErrDoesNotFit, strings.Join(exceededResources, ", "))
 	}
 
 	return nil
 }
 
+func (current *Resources) IsEmpty() bool {
+	return len(current.Flags) == 0 &&
+		len(current.Counts) == 0 &&
+		len(current.PersistentCounts) == 0 &&
+		len(current.Quantities) == 0
+}
+
 // NonEmptyString will return a string representation of the non-empty
 // Resources within the struct.
 func (current *Resources) NonEmptyString() string {
 	var resources []string
 
-	// Define function for checking int resources to keep code DRY
-	checkResource := func(resource string, value int) {
+	for c, value := range current.Counts {
 		if value > 0 {
-			resources = append(resources, resource)
+			resources = append(resources, fmt.Sprintf("%v:%v", c, value))
 		}
 	}
-
-	// Define function for checking quantity resources to keep code DRY
-	checkQuantityResource := func(resource string, currentVal resource.Quantity) {
-		if !currentVal.IsZero() {
-			resources = append(resources, resource)
+	for p, value := range current.PersistentCounts {
+		if value > 0 {
+			resources = append(resources, fmt.Sprintf("%v:%v", p, value))
 		}
 	}
-
-	checkResource("Apps", current.Apps)
-	checkResource("Containers", current.Containers)
-	checkResource("Jobs", current.Jobs)
-	checkResource("Volumes", current.Volumes)
-	checkResource("Secrets", current.Secrets)
-	checkResource("Images", current.Images)
-	checkResource("Projects", current.Projects)
-
-	checkQuantityResource("VolumeStorage", current.VolumeStorage)
-	checkQuantityResource("Memory", current.Memory)
-	checkQuantityResource("Cpu", current.CPU)
+	for q, value := range current.Quantities {
+		if !value.IsZero() {
+			resources = append(resources, fmt.Sprintf("%v:%v", q, value))
+		}
+	}
+	for p, value := range current.Flags {
+		resources = append(resources, fmt.Sprintf("%v:%v", p, value))
+	}
 
 	return strings.Join(resources, ", ")
 }
 
 // Equals will check if the current Resources struct is equal to another. This is useful
 // to avoid needing to do a deep equal on the entire struct.
 func (current *Resources) Equals(incoming Resources) bool {
-	return current.Unlimited == incoming.Unlimited &&
-		current.Apps == incoming.Apps &&
-		current.Containers == incoming.Containers &&
-		current.Jobs == incoming.Jobs &&
-		current.Volumes == incoming.Volumes &&
-		current.Secrets == incoming.Secrets &&
-		current.Images == incoming.Images &&
-		current.Projects == incoming.Projects &&
-		current.VolumeStorage.Cmp(incoming.VolumeStorage) == 0 &&
-		current.Memory.Cmp(incoming.Memory) == 0 &&
-		current.CPU.Cmp(incoming.CPU) == 0
+	// Convert both structs to JSON and compare the JSON representations.
+	currentJSON, err1 := json.Marshal(current)
+	incomingJSON, err2 := json.Marshal(incoming)
+
+	if err1 != nil || err2 != nil {
+		return false // Unable to marshal, return false.
+	}
+
+	return string(currentJSON) == string(incomingJSON)
```

Comment:
Should `Resource{}` and `Resource{Flags: Flags{Unlimited: false}}` be considered equal?

They seem semantically equal to me, but I don't think this implementation would consider them to be equal.